From 27503352dc910a6e13214df02d26187851e4e883 Mon Sep 17 00:00:00 2001
From: Enrico Joerns <ejo@pengutronix.de>
Date: Sat, 10 Mar 2018 01:15:34 +0100
Subject: [PATCH 42/44] src/service: fix packing checksum.size into GVariant
 dict

requires to be cast to packed type to be properly inserted.

Signed-off-by: Enrico Joerns <ejo@pengutronix.de>
---
 src/service.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/service.c b/src/service.c
index 9d83981..b61928d 100644
--- a/src/service.c
+++ b/src/service.c
@@ -238,7 +238,7 @@ static GVariant* convert_slot_status_to_dict(RaucSlot *slot) {
 
 	if (slot_state->checksum.digest && slot_state->checksum.type == G_CHECKSUM_SHA256) {
 		g_variant_dict_insert(&dict, "sha256", "s", slot_state->checksum.digest);
-		g_variant_dict_insert(&dict, "size", "t", slot_state->checksum.size);
+		g_variant_dict_insert(&dict, "size", "t", (guint64) slot_state->checksum.size);
 	}
 
 	if (slot_state->installed_timestamp) {
-- 
2.14.1

